include/beman/execution26/detail/async_concurrent_queue.hpp

```diff
@@ -0,0 +1,24 @@
+// include/beman/execution26/detail/async_concurrent_queue.hpp        -*-C++-*-
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef INCLUDED_BEMAN_EXECUTION26_DETAIL_ASYNC_CONCURRENT_QUEUE
+#define INCLUDED_BEMAN_EXECUTION26_DETAIL_ASYNC_CONCURRENT_QUEUE
+
+#include <beman/execution26/execution.hpp>
+#include <beman/execution26/detail/basic_concurrent_queue.hpp>
+#include <beman/execution26/detail/sender_of.hpp>
+#include <utility>
+
+// ----------------------------------------------------------------------------
+
+namespace beman::execution26::detail {
+template <typename T, typename Q>
+concept async_concurrent_queue = ::beman::execution26::detail::basic_concurrent_queue<T, Q> && requires(Q q, T&& t) {
```

Comment:
should this concept depend on an receivers environment?

Reply:
I have a long list of comments on the paper ([P0260r11](https://wg21.link/p0260r11)). I did think briefly about the same thing but came to the conclusion that it doesn't actually work to depend on the environment: the environment isn't known until the created senders are actually used. The operations might use <code>_sender-in-of</code> to include the environment. However, the actual interface is still up in the air and there is a chance that there are multiple value completions (e.g., `set_value_t(T)` and `set_value_t(conqueue_closed)` where `conqueue_closed` is type used to indicate that the queue was closed).

Reply:
I'm not convinced yet. I can imagine an async concurrent queue that requires a delegation scheduler in the environment. In my understanding it makes sense to ask, "are the async member functions valid senders if this environment is provided".
For the other point, I would have expected that error codes map into error completions. But I should direct those questions not to you but to the authors, maybe. I will try to join the next meetings, if possible.

Reply:
I think I can agree with that direction. I'm not sure how to express that or whether there is much point: the concepts are "exposition-only" and they are current not used at all! Thus, it is unclear whether they should even be there.